include/partitioner/cell_storage.hpp

```diff
@@ -52,11 +52,9 @@ namespace detail
 template <storage::Ownership Ownership> class CellStorageImpl
 {
   public:
-    using ValueOffset = std::uint32_t;
-    using BoundaryOffset = std::uint32_t;
+    using ValueOffset = std::size_t;
```

Comment:
Still an overflow risk if building on 32 bit machine, but I expect it would fail earlier on for planet sized data.

Reply:
Hmm, why not `std::uint64_t`?  I think, in general, when we're dealing with values that end up serialized, we should be explicit about sizes.

Reply:
The intention was to make it clear that this type represents an index into the other cell arrays.
I had not considered the serialization aspect though, so I'll change to `std::uint64_t` - it will also prevent any possible issue on 32 bit builds.